src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicConfiguration.cs

```diff
@@ -85,7 +85,7 @@ public static MsQuicSafeHandle Create(QuicServerConnectionOptions options, strin
         }
 
         X509Certificate? certificate = null;
-        X509Certificate[]? intermediates = null;
+        ReadOnlySpan<X509Certificate2> intermediates = ReadOnlySpan<X509Certificate2>.Empty;
```

Comment:
We should prefer `X509Certificate2` where possible.

Reply:
that one is tricky. Several of the underlying types in callbacks is `X509Certificate` for historical reasons. So we are possibly up to casting and cast exceptions. I'm not sure if it is worth it since all we do is serialization to MsQuic.

Reply:
BTW as far as the `default` I have no stone preference. The Empty span seems more explicit but either one should work